src/Metadata/Resource/Factory/ExtractorResourceMetadataCollectionFactory.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata\Resource\Factory;
+
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Delete;
+use ApiPlatform\Metadata\Extractor\ExtractorInterface;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\Patch;
+use ApiPlatform\Metadata\Post;
+use ApiPlatform\Metadata\Put;
+use ApiPlatform\Metadata\Resource\DeprecationMetadataTrait;
+use ApiPlatform\Metadata\Resource\ResourceMetadataCollection;
+use ApiPlatform\Tests\Fixtures\TestBundle\Entity\Comment;
+
+/**
+ * Creates a resource metadata from {@see Resource} extractors (XML, YAML).
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ * @experimental
+ */
+final class ExtractorResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
+{
+    use DeprecationMetadataTrait;
+    private $extractor;
+    private $decorated;
+    private $defaults;
+
+    public function __construct(ExtractorInterface $extractor, ResourceMetadataCollectionFactoryInterface $decorated = null, array $defaults = [])
+    {
+        $this->extractor = $extractor;
+        $this->decorated = $decorated;
+        $this->defaults = $defaults;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass): ResourceMetadataCollection
+    {
+        $resourceMetadataCollection = new ResourceMetadataCollection($resourceClass);
+        if ($this->decorated) {
+            $resourceMetadataCollection = $this->decorated->create($resourceClass);
+        }
+
+        if (Comment::class === $resourceClass) {
+            dd(class_exists($resourceClass), $this->extractor->getResources());
```

Comment:
dd? :D

Reply:
Work in progress :wink: 
The Behat tests and the console never go through this point, even if the Comment entity is enabled